src/utils/basicOperations.ts

```diff
@@ -1,10 +1,21 @@
 export const BZERO = BigInt(0);
 export const BONE = BigInt(1);
+export const ONE = BigInt('1000000000000000000'); // 18 decimal places
 
 const _require = (b: boolean, message: string) => {
     if (!b) throw new Error(message);
 };
 
+/**
+ * @dev Returns a scaling factor that, when multiplied to a token amount for `token`, normalizes its balance as if
+ * it had 18 decimals.
+ */
+export function _computeScalingFactor(tokenDecimals: bigint): bigint {
+    // Tokens with more than 18 decimals are not supported.
+    const decimalsDifference = BigInt(18) - tokenDecimals;
+    return ONE * BigInt(10) ** decimalsDifference;
```

Comment:
I'm having a hard time to follow this math here 😕 
Considering the USDC case with 6 decimals, `decimalsDifference` would be 12 and the return value would be 1e(18+12), right?
Shouldn't it be the opposite? Maybe `decimalsDifference` should be negative in the last line?
I believe this will be easier to double check after I replicate the helpers on other pools, where I'm able to test against this.
Please let me know if I'm missing something.

Reply:
Actually this was just taken directly from [SC](https://github.com/balancer-labs/balancer-v2-monorepo/blob/506897ea1e76c324d0b65985eb3dd2feb9e24959/pkg/pool-utils/contracts/BasePool.sol#L614). From my initial tests I believe we are using it correctly but worth remembering if you come across strange results.

Reply:
I tried looking into it and it seems that these maths are using more than 18 decimals internally and that's why this scalingFactor is greater than 18 decimals. I didn't go too deep, but the results are within expected, so I'll believe it's correct 😅